cmd/node/factory/structs.go

```diff
@@ -1502,13 +1438,16 @@ func generateGenesisHeadersAndApplyInitialBalances(
 			return nil, err
 		}
 
-		newStore, newBlkc, newMetaDataPool, err := createInMemoryStoreBlkcAndMetaDataPool(newShardCoordinator)
+		newStore, newBlkc, err := createInMemoryStoreBlkc(newShardCoordinator)
+		if err != nil {
+			return nil, err
+		}
 
 		argsMetaGenesis.ShardCoordinator = newShardCoordinator
 		argsMetaGenesis.Accounts = newAccounts
 		argsMetaGenesis.Store = newStore
 		argsMetaGenesis.Blkc = newBlkc
-		argsMetaGenesis.MetaDatapool = newMetaDataPool
+		argsMetaGenesis.DataPool = dataComponents.Datapool
```

Comment:
I would not use the same pointer of data pool

Reply:
done.